glean-core/src/database/mod.rs

```diff
@@ -495,15 +491,16 @@ impl Database {
             return Ok(());
         }
 
-        // Lifetime::Ping data is not persisted to disk if
-        // Glean has `delay_ping_lifetime_io` set to true
+        // Lifetime::Ping might also have data saved to `ping_lifetime_data`
+        // in case `delay_ping_lifetime_io` is set to true
         if lifetime == Lifetime::Ping {
             if let Some(ping_lifetime_data) = &self.ping_lifetime_data {
                 let mut data = ping_lifetime_data
                     .write()
                     .expect("Can't access app lifetime data as writable");
-                data.remove(&final_key);
-                return Ok(());
+                if data.remove(&final_key).is_some() {
+                    return Ok(());
```

Comment:
What if the data exists both in memory and on disk?  Shouldn't we clear it both places?  It looks like this will delete from memory, but leave it on disk.

Reply:
Data *should* never be in both places. Once I persist it I clear `ping_lifetime_data`, so it should either be in one place or the other.

But computers don't work everytime, so maybe we should check if it is not on both places... What do you think?

Reply:
I see.  That makes sense, as long as we can't multi-thread around this (but I don't think we can, since we require a single handle to the Glean object for everything we do).  I think we can ignore my suggestion for now and readdress only if we discover it's an issue.